Dockerfiles/pcluster-amazonlinux-2/Dockerfile

```diff
@@ -109,18 +109,23 @@ RUN curl -sOL https://efa-installer.amazonaws.com/aws-efa-installer-${EFA_INSTAL
     && cd .. \
     && rm -rf aws-efa-installer-${EFA_INSTALLER_VERSION}.tar.gz aws-efa-installer
 
-# Bootstrap spack compiler installation
-RUN mkdir -p /bootstrap && \
-    cd /bootstrap && \
-    git clone https://github.com/spack/spack spack \
-    && export SPACK_ROOT=/bootstrap/spack \
-    && . spack/share/spack/setup-env.sh \
-    && curl -sOL https://raw.githubusercontent.com/spack/spack-configs/main/AWS/parallelcluster/postinstall.sh \
-    && /bin/bash postinstall.sh -fg -nointel \
-    && spack clean -a \
-    && cd /bootstrap/spack \
-    && find . -type f -maxdepth 1 -delete \
-    && rm -rf bin lib share var /root/.spack
+# Bootstrap spack compiler installation into the eventual installation tree
+# Defined in spack/share/spack/gitlab/cloud_pipelines/configs/config.yaml
+ARG SPACK_ROOT="/bootstrap-compilers/spack"
+#  commit after https://github.com/spack/spack/pull/34821
+ARG SPACK_COMMIT=f27d012
+ARG TARGETARCH
+COPY Dockerfiles/pcluster-amazonlinux-2/packages-${TARGETARCH}.yaml /root/.spack/packages.yaml
+RUN mkdir -p $(dirname "${SPACK_ROOT}") \
+    && git clone https://github.com/spack/spack "${SPACK_ROOT}" \
+    && pushd "${SPACK_ROOT}" && git checkout ${SPACK_COMMIT} && popd \
+    && cp "${SPACK_ROOT}/share/spack/gitlab/cloud_pipelines/configs/config.yaml" "${SPACK_ROOT}/etc/spack/config.yaml" \
+    && . "${SPACK_ROOT}/share/spack/setup-env.sh" \
+    && spack compiler add \
+    && spack external find \
+    && spack tags build-tools | xargs -I {} spack config rm packages:{} \
```

Comment:
If I understand correctly, here we delete the build-tools we found. What are we looking for, exactly?

Reply:
Other externals such as `openssh`, `openssl`, etc.. If we include build-tools as externals we get version mismachtes when compiling packages, e.g. `gmp`.

Reply:
They are different per OS, pcluster version and when the user has run `yum update` the last time. I have been using this construct in `postinstall.sh` for a long time and it has worked well.

Reply:
> I have been using this construct in postinstall.sh for a long time and it has worked well.

The only worry is that this may change due to external choices, like tagging something as a "build-tool" in Spack (packages may have multiple tags). At that point it may be difficult to reconstruct the reason of the failure.